api_client/python/timesketch_api_client/crypto.py

```diff
@@ -0,0 +1,171 @@
+    """Class to store and retrieve stored credentials."""
+
+    # A default shared secret that will be used as part of the key.
+    SHARED_KEY = 'thetta er mjog leynt'
+
+    # The default filename of the token file.
+    DEFAULT_CREDENTIAL_FILENAME = '.timesketch.token'
+
+    # The number of characters in the random part of the default key.
+    RANDOM_KEY_LENGTH = 20
+
+    def __init__(self):
+        """Initialize the class."""
+        self._user = getpass.getuser()
+        home_path = os.path.expanduser('~')
+        self._filepath = os.path.join(
+            home_path, self.DEFAULT_CREDENTIAL_FILENAME)
+
+    def _get_key(self, seed_key):
+        """Returns an encryption key.
+
+        Args:
+            seed_key (str): a seed used to generate an encryption key.
+
+        Returns:
+            Bytes with the encryption key.
+        """
+
+        key_string_half = '{0:s}{1:s}'.format(
+            getpass.getuser(), seed_key)
+
+        if len(key_string_half) >= 32:
+            key_string = key_string_half[:32]
+        else:
+            key_string = '{0:s}{1:s}'.format(
+                key_string_half, self.SHARED_KEY)
+            key_string = key_string[:32]
+
+        return base64.b64encode(bytes(key_string, 'utf-8'))
+
+    def set_filepath(self, file_path):
+        """Set the filepath to the credential file."""
+        if os.path.isfile(file_path):
+            self._filepath = file_path
+
+    def save_credentials(self, cred_obj, file_path=''):
+        """Save credential data to a token file.
+
+        This function will create an encrypted file
+        in the user's home directory (~/.timesketch.token by default)
+        that contains a stored copy of the credential object.
+
+        Args:
+            cred_obj (google.oauth2.credentials.Credentials): the credential
+                object that is to be stored on disk.
+            file_path (str): Full path to the file storing the saved
+                credentials.
+        """
+        # TODO (kiddi): Support user/pass credentials as well. Create
+        # a separate credential object that wraps the OAUTH creds.
+        if not file_path:
+            file_path = self._filepath
+
+        if not os.path.isfile(file_path):
+            logger.info('File does not exist, creating it.')
+
+        data = {
+            'token': cred_obj.token,
+            '_scopes': getattr(cred_obj, '_scopes', []),
+            '_refresh_token': getattr(cred_obj, '_refresh_token', ''),
+            '_id_token': getattr(cred_obj, '_id_token', ''),
+            '_token_uri': getattr(cred_obj, '_token_uri', ''),
+            '_client_id': getattr(cred_obj, '_client_id', ''),
+            '_client_secret': getattr(cred_obj, '_client_secret', ''),
+        }
+        if cred_obj.expiry:
+            data['expiry'] = cred_obj.expiry.isoformat()
+        data_string = json.dumps(data)
+
+        letters = string.ascii_letters
+        random_string = ''.join(
```

Comment:
this is a seed key, right? make the name reflect this in that case

Reply:
done